### ISSUE TITLE
Incorrect predictions

### ISSUE THREAD

**Issue by sasinghi** [zhixuhao/unet#65]
After training the model (for a short time) I see an accuracy of 80%. However, when I test it, the result is as shown for all test images. Can someone help with this? Would be great if someone could share the pre-trained weights. 

![image](https://user-images.githubusercontent.com/9640216/44722088-3db32100-aacc-11e8-9643-749923867502.png)

**Comment by ajithvallabai** [comment]
change 
`model.fit_generator(myGene,steps_per_epoch=2000,epochs=5,callbacks=[model_checkpoint])`

in above line change the steps_per_epoch to 300 or 400 or 500 some thing should work

**Comment by enggrey** [comment]
In case you or anyone else is still having issues with this, in the testGenerator function change this line
`img = img / 255`
for this
`img = img / 255.`

**Comment by sasinghi** [comment]
@ajithvallabai that was it. Training longer helped. 
Thanks!

**Comment by duanmuji** [comment]
I modified img.255, but problems persisted.

**Comment by openXG** [comment]
> `model.fit_generator(myGene,steps_per_epoch=2000,epochs=5,callbacks=[model_checkpoint])`
> 
> in above line change the steps_per_epoch to 300 or 400 or 500 some thing should work

I set steps_per_epoch=500, but my test images are still all gray...
should I set it more larger than 500? by the way, I just use CPU.

**Comment by JianyingLi** [comment]
> > `model.fit_generator(myGene,steps_per_epoch=2000,epochs=5,callbacks=[model_checkpoint])`
> > in above line change the steps_per_epoch to 300 or 400 or 500 some thing should work
> 
> I set steps_per_epoch=500, but my test images are still all gray...
> should I set it more larger than 500? by the way, I just use CPU.

i think epochs should be increased rather than steps_per_epoch to improve the acc . Another place to check is the function of adjustData.
def adjustData(img,mask,flag_multi_class,num_class):
    # here#
    elif(np.max(img) > 1):
        img = img / 255
        mask = mask /255
        mask[mask > 0.5] = 1
        mask[mask <= 0.5] = 0
    return (img,mask)``
This code works for py3, if you use py2, modify '255' to '255.'; Last, you need to check the threshold '0.5' in order to make sure the returned mask is a binary(0/1) mask.

**Comment by lihao6666** [comment]
get it

**Comment by jizhang02** [comment]
@JianyingLi Hello,
I change cross entropy loss into dice, but the result become all white, could you help me?
def dice_coef(y_true, y_pred):
    smooth = 1
    y_true_f = K.flatten(y_true)
    y_pred_f = K.flatten(y_pred)
    intersection = K.sum(y_true_f * y_pred_f)
    return (2. * intersection +smooth) / (K.sum(y_true_f) + K.sum(y_pred_f) +smooth)

def dice_coef_loss(y_true, y_pred):
    print("dice loss")
    return 1-dice_coef(y_true, y_pred)

 model.compile(optimizer = Adam(lr = 1e-5), loss = dice_coef_loss, metrics = ['accuracy'])
